### PR TITLE
logging: fix up formatting initialization and remove duplicate timestamps

### PR DESCRIPTION
**Pull request by ianvernon**
With the addition of the default logger, formatting of logs was broken. Refactor
setupFormatter to return a logrus Formatter and use this formatter for the
default logger. Move logging initialization to the top of the cilium-agent's
initialization. Also stops logrus from adding its timestamp field to logs, as we
desire to have as clean and brief logs as possible; timestamping the logs is
left up to the logging framework to which Cilium logs (e.g., journalctl).

Signed-off by: Ian Vernon <ian@cilium.io>

Fixes: #1023

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
Once build finishes, will provide log snippets from both K8s and from Cilium when running as a systemd service.

**Review by manalibhutiyani — APPROVED** [review]

File: pkg/logging/logging.go [hunk 1]
```diff
@@ -104,12 +104,16 @@ func setFireLevels(level logrus.Level) []logrus.Level {
 	}
 }
 
+// InitializeDefaultLogger returns a logrus Logger with a custom text formatter.
+func InitializeDefaultLogger() *logrus.Logger {
```
Comment (raybejjani) [inline pkg/logging/logging.go:108]: Could you also make a call to this at https://github.com/cilium/cilium/blob/master/pkg/endpoint/log.go#L62 so it also gets the correct formatter?
The endpoints end up with a separate logger object when their config is set to debug. In hindsight, a better solution to that problem would be a global DefaultDebugLogger that would be here, but that isn't the point of your PR (but feel free to do it that way or any way you feel is best).

**Comment by ianvernon** [comment]
Excerpt of logs from cilium pod running in K8s:
```
level=debug msg="Considering removing iptables rule" obj="-N DOCKER"
level=debug msg="Considering removing iptables rule" obj="-N DOCKER-ISOLATION"
level=debug msg="Considering removing iptables rule" obj="-N KUBE-FIREWALL"
level=debug msg="Considering removing iptables rule" obj="-N KUBE-SERVICES"
level=debug msg="Considering removing iptables rule" obj="-A INPUT -m comment --comment \"kubernetes service portals\" -j KUBE-SERVICES"
level=debug msg="Considering removing iptables rule" obj="-A INPUT -j KUBE-FIREWALL"
level=debug msg="Considering removing iptables rule" obj="-A FORWARD -j DOCKER-ISOLATION"
level=debug msg="Considering removing iptables rule" obj="-A FORWARD -o docker0 -j DOCKER"
level=debug msg="Considering removing iptables rule" obj="-A FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT"
level=debug msg="Considering removing iptables rule" obj="-A FORWARD -i docker0 ! -o docker0 -j ACCEPT"
level=debug msg="Considering removing iptables rule" obj="-A FORWARD -i docker0 -o docker0 -j ACCEPT"
level=debug msg="Considering removing iptables rule" obj="-A OUTPUT -m comment --comment \"kubernetes service portals\" -j KUBE-SERVICES"
level=debug msg="Considering removing iptables rule" obj="-A OUTPUT -j KUBE-FIREWALL"
level=debug msg="Considering removing iptables rule" obj="-A DOCKER-ISOLATION -j RETURN"
level=debug msg="Considering removing iptables rule" obj="-A KUBE-FIREWALL -m comment --comment \"kubernetes firewall for dropping marked packets\" -m mark --mark 0x8000/0x8000 -j DROP"
level=debug msg="Considering removing iptables rule" obj="-A KUBE-SERVICES -d 172.20.0.10/32 -p udp -m comment --comment \"kube-system/kube-dns:dns has no endpoints\" -m udp --dport 53 -j REJECT --reject-with icmp-port-unreachable"
level=debug msg="Considering removing iptables rule" obj="-A KUBE-SERVICES -d 172.20.0.10/32 -p tcp -m comment --comment \"kube-system/kube-dns:dns-tcp has no endpoints\" -m tcp --dport 53 -j REJECT --reject-with icmp-port-unreachable"
```

Excerpt of Cilium logs running as systemd service on node directly:
```
Jan 16 16:23:12 cilium-master systemd[1]: Started cilium.
Jan 16 16:23:12 cilium-master cilium-agent[12122]: level=debug msg="os.Hostname() returned" nodeName=cilium-master
Jan 16 16:23:12 cilium-master cilium-agent[12122]: level=info msg="     _ _ _"
Jan 16 16:23:12 cilium-master cilium-agent[12122]: level=info msg=" ___|_| |_|_ _ _____"
Jan 16 16:23:12 cilium-master cilium-agent[12122]: level=info msg="|  _| | | | | |     |"
Jan 16 16:23:12 cilium-master cilium-agent[12122]: level=info msg="|___|_|_|_|___|_|_|_|"
Jan 16 16:23:12 cilium-master cilium-agent[12122]: level=info msg="Cilium 1.0.0-rc3 e2b0802 2018-01-17T00:06:30+00:00 go version go1.9 linux/amd64"
Jan 16 16:23:12 cilium-master cilium-agent[12122]: level=info msg="clang (3.8.1) and kernel (4.9.17) versions: OK!"
Jan 16 16:23:12 cilium-master cilium-agent[12122]: level=info msg="linking environment: OK!"
Jan 16 16:23:12 cilium-master cilium-agent[12122]: level=info msg="bpf_requirements check: OK!"
Jan 16 16:23:12 cilium-master cilium-agent[12122]: level=debug msg="bpf_features check: Some features may be limited:"
Jan 16 16:23:12 cilium-master cilium-agent[12122]: level=debug msg="BPF/probes: CONFIG_CGROUP_BPF=y is not in kernel configuration"
Jan 16 16:23:12 cilium-master cilium-agent[12122]: level=debug msg="BPF/probes: CONFIG_LWTUNNEL_BPF=y is not in kernel configuration"
Jan 16 16:23:12 cilium-master cilium-agent[12122]: level=debug msg="HAVE_LPM_MAP_TYPE: Your kernel doesn't support LPM trie maps for BPF, thus disabling CIDR policies. Recommendation is to run 4.11+ kernels."
Jan 16 16:23:12 cilium-master cilium-agent[12122]: level=debug msg="HAVE_LRU_MAP_TYPE: Your kernel doesn't support LRU maps for BPF, thus switching back to using hash table for the cilium connection tracker. Recommendation is to run 4.10+ kernels."
Jan 16 16:23:12 cilium-master cilium-agent[12122]: level=info msg="Waiting for consul client to be ready..."
```

These are achieved via running `journalctl` - note no duplicate timestamp.

File: pkg/logging/logging.go (hunk 1 above)
Comment (ianvernon) [reply pkg/logging/logging.go:108]: Utilized this function at the line as you requested.

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
Ginkgo tests failed due to an issue which will be fixed by #2563 .

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
I am unable to reopen this PR because the code was merged by myself accidentally (d'oh). I reverted the changes, merged the revert, and reopened another PR with these changes at #2566. Sorry for the confusion :(

**Pull request merged**